jacoco.gradle

```
@@ -0,0 +1,48 @@
+apply plugin: "jacoco"
+jacoco {
+    toolVersion = "0.8.5"
+}
+
+test {
+    jacoco {
+        excludes = []
+    }
+//    finalizedBy jacocoTestReport // report is always generated after tests run
+        finalizedBy jacocoTestCoverageVerification // code coverage is verified after tests run
+}
+jacocoTestReport {
+    dependsOn test // tests are required to run before generating the report
+    reports {
+        xml.enabled false
+        csv.enabled false
+        html.destination file("${buildDir}/reports/jacoco")
+    }
+    afterEvaluate {
+        classDirectories.setFrom(files(classDirectories.files.collect {
+            fileTree(dir: it, exclude: [
+            ])
+        }))
+    }
+}
+
+jacocoTestCoverageVerification {
+    dependsOn jacocoTestReport
+    afterEvaluate {
+        classDirectories.setFrom(files(classDirectories.files.collect {
+            fileTree(dir: it, exclude: [
+            ])
+        }))
+    }
+    violationRules {
+        rule {
+            limit {
+                minimum = 0.5 // verifying mandatory code coverage of 50%
+            }
+        }
+        rule {
+            element = 'METHOD'
+            excludes = [
+            ]
+        }
+    }
+}
```

new line here as well.
```suggestion
}

```